mapiproxy/servers/default/emsmdb/oxcmsg.c

```diff
@@ -1485,7 +1486,9 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopCreateAttach(TALLOC_CTX *mem_ctx,
 								   messageID, message_object);
 		if (attachment_object) {
 			mretval = mapistore_message_create_attachment(emsmdbp_ctx->mstore_ctx, contextID, message_object->backend_object,
-								      attachment_object, &attachment_object->backend_object, &mapi_repl->u.mapi_CreateAttach.AttachmentID);
+								      attachment_object, &attachment_object->backend_object, &attachmentID);
+			attachment_object->object.attachment->attachmentID = attachmentID;
```

Comment:
The specification does not say nothing about the attachmentId value in case of error.

I have chosen this layout only to parallel the one in CreateMessage and to have the less change in behaviour possible. However, like the last Julien's change in CreateMessage the AttachmentID should be initialized to avoid problems. I will assign it to 0 (first valid attachment id|)